### PR TITLE
L3-66 Troubleshoot integration between middleware and goldi SSO

### PR DESCRIPTION
**Pull request by mgwozdz-unicon**
## Description
Fixed issues related to differences between local and AWS environments as well as with the integration.
- Updated WebSecurityConfig to only send LTI requests through the LTI oAuth Filter. Canvas made several additional irrelevant requests that would break the LTI flow if they went through this filter because they would not contain the expected state.
- Added properties to the properties file so that Spring would include the Secure flag in the `set-cookie` header. Without this flag, the browser would refuse to set the jsessionid cookie, which is what Java uses to track the session, which is where the state is stored. Therefore, without the Secure cookie flag, the LTI state would get lost after the OIDC initiation.
- Used client side javascript to be able to perform a hard redirect POST to Goldilocks, ensuring that the UI displays properly.
- Updated unit tests accordingly and added instructions to the README on how to test in the AWS dev environment.

### Motivation and Context
These changes are necessary for the LTI integration between the middleware and Goldilocks to function.

### Fixes
[L3-66](https://lumenlearning.atlassian.net/browse/L3-66)

## How Has This Been Tested?
0. Register the middleware with Canvas and import the appropriate Common Cartridge into a new Canvas course.
1. Click on the links generated by the Common Cartridge.
2. Ensure that the study plan is visible, it is possible to navigate through the Dive In PBJ pages, and that it is possible to take quizzes.

---

## Checklist
- [x] I have reviewed the AC for this feature and my changes meet all requirements
- [x] (For UI changes) I have considered the accessibility of this feature (see https://www.a11yproject.com/checklist/ for a high-level checklist)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have requested a review from at least one other dev

### CODE REVIEW

File: src/main/java/net/unicon/lti/security/lti/LTI3OAuthProviderProcessingFilter.java [hunk 1]
```diff
@@ -68,76 +71,87 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
...
-                log.debug("Exception " + ex.getMessage(), ex);
-                ((HttpServletResponse) servletResponse).setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-            } catch (DataServiceException e) {
-                log.error("Error in the Data Service", e);
+            // First we validate that the state is a good state.
+
+            //First, we make sure that the query has a state
+            String state = httpServletRequest.getParameter("state");
+            String link = httpServletRequest.getParameter("link");
+            if (httpServletRequest.getSession().getAttribute("lti_state") == null) {
```
Comment (smith750) [inline src/main/java/net/unicon/lti/security/lti/LTI3OAuthProviderProcessingFilter.java:106]: Is this okay or was this the thing they're moving, long term, into local storage hosted by the LMS?

**Review by smith750 — APPROVED** [review]
This looks good though I am curious about the session stuff - but we can fix that over time.  Thank you so much, Mary!

**Review by leannethng — APPROVED** [review]
Thanks again for all your comments and documentation! I have tested this branch through the dev-env as it is currently what is running there and it all works as expected on Chrome and Firefox for me.

File: src/main/java/net/unicon/lti/security/lti/LTI3OAuthProviderProcessingFilter.java (hunk 1 above)
Comment (mgwozdz-unicon) [reply src/main/java/net/unicon/lti/security/lti/LTI3OAuthProviderProcessingFilter.java:106]: The state is what will ultimately get saved in an iframe hosted by the LMS as part of the cookie solution down the road. Canvas (and other LMSs) haven't yet implemented the LMS storage solution, but it is coming soon.

Comment (smith750) [reply src/main/java/net/unicon/lti/security/lti/LTI3OAuthProviderProcessingFilter.java:106]: Yep - thank you for the full explanation there!

**Pull request merged**